tests/correctness/resources/miniboxing/tests/compile/gh-bug-214.scala

```diff
@@ -0,0 +1,37 @@
+import scala.reflect._
+
+object Test {
+  def baz[@miniboxed T : ClassTag](t: T) = {
+    new Array[T](42) // should warn
+    Array[T](t,t,t,t,t) // should warn
+  }
+  def bar[T : ClassTag](t: T) = {
+    new Array[T](42)
+    Array[T](t,t,t,t,t)
+    new Array[Int](42)
+  }
+
+  def foo[@miniboxed T](t: T) = {
+    MbArray[T](t, t, t)
+  }
```

Comment:
```
scala> def foo[T](t: T) = MbArray[T](t, t, t, t)
<console>:10: warning: The following code instantiating an MbArray object cannot be optimized since the type argument is not a primitive type (like Int), a miniboxed type parameter or a subtype of AnyRef. This means that primitive types could end up boxed:
       def foo[T](t: T) = MbArray[T](t, t, t, t)
                          ^
foo: [T](t: T)MbArray[T]

scala> def foo[T](t: T) = new MbArray[T](10)
<console>:10: error: class MbArray is abstract; cannot be instantiated
       def foo[T](t: T) = new MbArray[T](10)
                          ^
```